include/openssl/macros.h

```diff
@@ -58,14 +60,18 @@
           __attribute__((deprecated("Since OpenSSL " # since)))
 #     define OSSL_DEPRECATED_FOR(since, message) \
           __attribute__((deprecated("Since OpenSSL " # since ";" message)))
+#     define OSSL_PRIVATE_MP_DEPRECATED_PRAGMA(s) _Pragma(#s)
+#     define OSSL_DEPRECATED_PRAGMA(s) OSSL_PRIVATE_MP_DEPRECATED_PRAGMA(GCC warning s)
 #    elif __GNUC__ > 3 || (__GNUC__ == 3 && __GNUC_MINOR__ > 0)
 #     define OSSL_DEPRECATED(since) __attribute__((deprecated))
 #     define OSSL_DEPRECATED_FOR(since, message) __attribute__((deprecated))
+#     define OSSL_DEPRECATED_PRAGMA(s)
 #    endif
 #   elif defined(__SUNPRO_C)
 #    if (__SUNPRO_C >= 0x5130)
 #     define OSSL_DEPRECATED(since) __attribute__ ((deprecated))
 #     define OSSL_DEPRECATED_FOR(since, message) __attribute__ ((deprecated))
+#     define OSSL_DEPRECATED_PRAGMA(s)
```

Comment:
You will need some empty fallback ` OSSL_DEPRECATED_PRAGMA(s)` for all other platforms and compilers. Similarly as done for OSSL_DEPRECATED below.

Reply:
Fine like that?

Reply:
Yeah, better.